DOCSUP-11585: Document Date32 data type (#27919)
ClickHouse/ClickHouse: docs/en/sql-reference/functions/type-conversion-functions.md

## toDateTimeOrNull {#todatetimeornull}

> ## toDate32 {#toDate32}

Reviewer: ```suggestion
## toDate32 {#todate32}
```

> Converts the argument to the [Date32](../../sql-reference/data-types/date32.md) data type. If the value is outside the range returns the border values of [Date32](../../sql-reference/data-types/date32.md). If the argument has [Date](../../sql-reference/data-types/string.md) type borders of `Date` are taken into account.

Reviewer: ```suggestion
Converts the argument to the [Date32](../../sql-reference/data-types/date32.md) data type. If the value is outside the range returns the border values supported by `Date32`. If the argument has [Date](../../sql-reference/data-types/string.md) type, borders of `Date` are taken into account.
```

**Syntax** 

``` sql
toDate32(expr)
```

**Arguments** 

> -   `expr` — The value. [String](../../sql-reference/data-types/string.md), [UInt32](../../sql-reference/data-types/int-uint.md) or [Date](../../sql-reference/data-types/string.md).

Reviewer: ```suggestion
-   `expr` — The value. [String](../../sql-reference/data-types/string.md), [UInt32](../../sql-reference/data-types/int-uint.md) or [Date](../../sql-reference/data-types/date.md).
```

**Returned value**

-   A calendar date.

Type: [Date32](../../sql-reference/data-types/date32.md).

**Example**

> 1. If the value is within the range:

Reviewer: ```suggestion
1. The value is within the range:
```

``` sql
SELECT toDate32('1955-01-01') AS value, toTypeName(value);
```

``` text
┌──────value─┬─toTypeName(toDate32('1925-01-01'))─┐
│ 1955-01-01 │ Date32                             │
└────────────┴────────────────────────────────────┘
```

> 2. If the value is outside the range:

Reviewer: ```suggestion
2. The value is outside the range:
```

``` sql
SELECT toDate32('1924-01-01') AS value, toTypeName(value);
```

``` text
┌──────value─┬─toTypeName(toDate32('1925-01-01'))─┐
│ 1925-01-01 │ Date32                             │
└────────────┴────────────────────────────────────┘
```

3. With `Date`-type argument:

``` sql
SELECT toDate32(toDate('1924-01-01')) AS value, toTypeName(value);
```

``` text
┌──────value─┬─toTypeName(toDate32(toDate('1924-01-01')))─┐
│ 1970-01-01 │ Date32                                     │
└────────────┴────────────────────────────────────────────┘
```

> ## toDate32OrZero {#toDate32OrZero}

Reviewer: ```suggestion
## toDate32OrZero {#to-date32-or-zero}
```

> The same as `toDate32` but returns the min value of [Date32](../../sql-reference/data-types/date32.md) if a zero argument is received.

Reviewer 2: ```suggestion
The same as `toDate32` but returns the min value of [Date32](../../sql-reference/data-types/date32.md) if invalid argument is received.
```

> The same as `toDate32` but returns the min value of [Date32](../../sql-reference/data-types/date32.md) if invalid argument is received.

Reviewer: ```suggestion
The same as [toDate32](#toDate32) but returns the min value of [Date32](../../sql-reference/data-types/date32.md) if invalid argument is received.
```

**Example**

Query:

``` sql
SELECT toDate32OrZero('1924-01-01'), toDate32OrZero('');
```

Result:

``` text
┌─toDate32OrZero('1924-01-01')─┬─toDate32OrZero('')─┐
│                   1925-01-01 │         1925-01-01 │
└──────────────────────────────┴────────────────────┘
```

> ## toDate32OrNull {#todate32OrNull}

Reviewer: ```suggestion
## toDate32OrNull {#to-date32-or-null}
```

> The same as `toDate32` but returns `NULL` if invalid argument is received.

Reviewer: ```suggestion
The same as [toDate32](#toDate32) but returns `NULL` if invalid argument is received.
```
